carbonmark/components/pages/Project/Retire/Pool/RetireInputs.tsx

```diff
@@ -188,6 +188,10 @@ export const RetireInputs: FC<Props> = (props) => {
       : props.product.leftToSell
   );
 
+  const isICR =
+    props.product.type === "listing" &&
```

Comment:
Why do we need this check? I get the id / registry check but why does pool vs listing matter?

Reply:
If this check is omitted a type error pops up because `project` doesn't exist on the "pool" `product` type.

Reply:
@psparacino @sprrwhwk following the prop drilling back up to /projects/retire/[lsitingId] or /projects/retire/[pool] it looks like project is coming from 

https://github.com/KlimaDAO/klimadao/blob/b784b79139e78b177db2dd6fd0dbbc772523f11e/carbonmark/.generated/carbonmark-api-sdk/clients/getProjectsId.ts#L13

Can we use the server types here instead of the handcrafted ones?

Reply:
Ok moving this conversation to a new ticket

https://github.com/KlimaDAO/klimadao/issues/2293